MapboxDirections/OfflineDirections.swift

```diff
@@ -14,20 +14,22 @@ struct AvailableVersionsResponse: Codable {
 public protocol OfflineDirectionsProtocol {
     
     /**
-     Fetches the available versions.
+     Fetches the available routing tile versions. Typically, the most recent version should be passed into `downloadTiles(in:version:completionHandler:)`.
+
+     - parameter completionHandler: Executes when the download completes or fails.
      */
-    @discardableResult
-    func fetchAvailableOfflineVersions(completionHandler: @escaping OfflineVersionsHandler) -> URLSessionDataTask
+    @objc(fetchAvailableOfflineVersionsWithCompletion:)
+    func fetchAvailableOfflineVersions(completionHandler: @escaping OfflineVersionsHandler) -> ()
     
     /**
-     Initiates a download process of all tiles needed to provide routing within the given bounding box.
+     Fetches routing tiles needed to generate a route within the given bounding box.
      
      - parameter coordinateBounds: The region of the pack to be downloaded.
      - parameter version: The version of the pack to be downloaded.
-     - parameter completionHandler: Informs when the download is completed or failed. The offline pack may be moved from the temporary directory and to a persistent store at this point.
+     - parameter completionHandler: Executes when the download completes or fails. The offline pack may be moved from the temporary directory to a persistent store at this point.
      */
-    @discardableResult
-    func downloadTiles(in coordinateBounds: CoordinateBounds, version: OfflineVersion, session: URLSession?, completionHandler: @escaping OfflineDownloaderCompletionHandler) -> URLSessionDownloadTask
+    @objc(downloadTilesInCoordinateBounds:withVersion:completion:)
```

Comment:
Nit: the usual parameter name in Objective-C is either `completionHandler` or `completionBlock`.

Reply:
really? OK, let's go with `completionHandler` since that's what's in the swift